docs/developer-guide/writing-epics.md

```diff
@@ -0,0 +1,221 @@
+
+An **epic** is basically nothing more that:
+
+ *a function that returns a stream of redux actions to emit.*
+
+A simple epic in mapstore can be this:
+
+```javascript
+const fetchUserEpic = (action$, store) => action$
+    .ofType(MAP_CONFIG_LOADED)
+    .filter(() => isMapLoadConfigurationEnabled(store.getState()))
+    .map({
+        type: NOTIFICATION,
+        message: "Map Loaded"
+    });
+);
+```
+
+The epic function has 2 arguments:
+
+- `action$`: the stream of of redux action. Every time an action is triggered redux, it is emitted as an event on the `action$` stream.
+- `store`. A *small version* of redux store, that contains essentially only one important method called `getState()`. This method returns the current redux state object.
+
+This function **must return** a new stream that emits the action we want to send back to redux. The **redux-observable** middleware subscribes the epics returned streams so the action will be automatically triggered on redux when they are emitted by the stream.
+
+> **NOTE**: **redux-observable** middleware is already added to the MapStore2's StandardStore and StandardApp, so a developer should only take care of creating his own epics and add them to MapStore.
+
+Typically the stream returned by an epic is always listening for new actions and sends other actions:
+
+> ** actions in, actions out **.
+
+Let's analyze the epic reported as first example:
+
+It returns a stream ( arrow function (`=>`) implicit return) created manipulating the `action$` stream. It first filters out all the unwanted actions catching only the `MAP_CONFIG_LOADED` action types, then another filter checks the state to verify some condition (typically a `selector` can be used to check the state).
+
+> **NOTE**: **redux-observable** adds an operator to rxjs called `ofType` that simply filters the action of certain types, passed as argument, but it is not a part of standard RxJS operators
+
+At the end, the events that pass the 2 filters, pass by the `map` operator. The map operator simply returns the (action) object:
+
+```javascript
+{
+    type: NOTIFICATION,
+    message: "Map Loaded"
+}
+```
+
+This object will be emitted on the returning stream and so the action will be triggered in redux.
+
+Of course instead of emitting the plain object, you can use an action creator, like this:
+
+```javascript
+const notifyMapLoaded = (action$, store) => action$
+    .ofType(MAP_CONFIG_LOADED)
+    .filter(() => isMapLoadConfigurationEnabled(store.getState()))
+    .map(info({
+        message: "Map Loaded"
+    }))
+);
+```
+
+## Doing async
+
+Typical async operation involve operators like:
+
+- [switchMap](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-switchMap)
+- [mergeMap](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-mergeMap)
+- many others
+
+The base concept of all these solutions is to project to the main stream the result of some other observable.
```

Comment:
I would explain the meaning of "project"

Reply:
I changed a little bit the purpose of this paragraph and so this phrase, check if does it seem better